include/dem/disable_particle_contact.h

```diff
@@ -0,0 +1,246 @@
+
+  /**
+   * Carries out the calculation of the granular temperature and solid
+   * fraction approximation in each local cell. Those values are a criteria
+   * for cell mobility
+   *
+   * @param triangulation The triangulation
+   * @param particle_handler The particle handler that contains all the particles
+   */
+  void
+  calculate_cell_granular_temperature(
+    const DoFHandler<dim> &                background_dh,
+    const Particles::ParticleHandler<dim> &particle_handler);
+
+  /**
+   * Find the mobility status of a cell
+   *
+   * @param cell The iterator of the cell that needs mobility evaluation
+   */
+  inline unsigned int
+  check_cell_mobility(
+    const typename Triangulation<dim>::active_cell_iterator &cell) const
+  {
+    unsigned int status;
+    // Check if the cell is in the mobile status set
+    status = mobility_status::mobile;
+    auto cell_iterator_from_mobile_container =
+      status_to_cell[status].find(cell);
+
+    if (cell_iterator_from_mobile_container != status_to_cell[status].end())
+      {
+        return status;
+      }
+
+    // Check if the cell is in the active status set
+    status = mobility_status::active;
+    auto cell_iterator_from_active_container =
+      status_to_cell[status].find(cell);
+
+    if (cell_iterator_from_active_container != status_to_cell[status].end())
+      {
+        return status;
+      }
+
+    // If not mobile or active, cell is inactive
+    return mobility_status::inactive;
+  }
+
+  /**
+   * Convert the vector of mobility status set to a vector of mobility status
+   * with cell id order
+   * status_to_cell is a vector of 3 sets of cell iterator (inactive, active,
+   * mobile), it can't be used as is in the pvd post-processing or any data out,
+   * it needs to be converted to a vector of mobility status by active cell
+   * index
+   *
+   * @param status The initiated vector for the conversion
+   */
+  void
+  get_mobility_status_vector(std::vector<unsigned int> &status)
+  {
+    // Loop over all set with different mobility status (except empty)
+    for (unsigned int i_set = 0; i_set < mobility_status::n_mobility_status - 1;
+         i_set++)
+      {
+        for (auto &cell : status_to_cell[i_set])
+          {
+            if (cell->is_locally_owned())
+              {
+                status[cell->active_cell_index()] = i_set;
+              }
+          }
+      }
+  };
+
+  void
+  get_mobility_status_vector(Vector<float> &status)
+  {
+    // Loop over all set with different mobility status (except empty)
+    for (unsigned int i_set = 0; i_set < mobility_status::n_mobility_status - 1;
+         i_set++)
+      {
+        for (auto &cell : status_to_cell[i_set])
+          {
+            if (cell->is_locally_owned())
+              {
+                status[cell->active_cell_index()] = i_set;
+              }
+          }
+      }
+  };
+
+  std::vector<typename DEM::dem_data_structures<dim>::cell_set>
+  get_mobility_status()
```

Comment:
if you have two containers for this information, I would use get_mobility_status_map()